hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChunkBuffer.java

```diff
@@ -52,6 +57,30 @@ static ChunkBuffer allocate(int capacity, int increment) {
     return new ChunkBufferImplWithByteBuffer(ByteBuffer.allocate(capacity));
   }
 
+  /** Preallocate the entire buffer. */
+  static ChunkBuffer preallocate(long capacity, int increment) {
+    Preconditions.assertTrue(increment > 0);
+    if (capacity <= increment) {
+      final CodecBuffer c = CodecBuffer.allocateDirect(Math.toIntExact(capacity));
```

Comment:
Would be cleaner if we directly deal with `ByteBufAllocator` and  `ByteBuf` in ChunkBuffer. `CodecBuffer` logic doesn't provide much, but adds an unnecessary dependency.